src/Formik.tsx

```diff
@@ -259,92 +253,91 @@ export function useFormik<Values = object>({
-      } else {
-        return Promise.resolve({});
-      }
+  // Are there any validation methods?
+  const hasValidationFns = React.useMemo(
+    () =>
+      props.validationSchema ||
+      props.validate ||
+      (fields.current &&
+        Object.keys(fields.current).filter(
+          key => !!fields.current[key].validate
+        ).length > 0),
+    [props.validate, props.validationSchema, fields]
+  );
+
+  // Run all validations and return the result
+  const runAllValidations = React.useCallback(
+    (values: Values) => {
+      return Promise.all([
+        runFieldLevelValidations(values),
+        props.validationSchema ? runValidationSchema(values) : {},
+        props.validate ? runValidateHandler(values) : {},
+      ]).then(([fieldErrors, schemaErrors, validateErrors]) => {
+        const combinedErrors = deepmerge.all<FormikErrors<Values>>(
+          [fieldErrors, schemaErrors, validateErrors],
+          { arrayMerge }
+        );
+        return combinedErrors;
+      });
     },
     [
       props.validate,
       props.validationSchema,
       runFieldLevelValidations,
       runValidateHandler,
       runValidationSchema,
-      state.errors,
-      state.values,
-      fields,
     ]
   );
 
-  React.useEffect(() => {
-    if (
-      prevState.values !== state.values &&
-      !!validateOnChange &&
-      !state.isSubmitting &&
-      isMounted.current != null
-    ) {
-      const [validate, cancel] = makeCancelable(validateForm());
-      validate.then(x => x).catch(x => x); // catch the rejection silently
-      return cancel;
-    }
-    return;
-  }, [
-    prevState.values,
-    state.isSubmitting,
-    state.values,
-    validateForm,
-    validateOnChange,
-    isMounted,
-  ]);
+  // Run validations and dispatching the result as low-priority via rAF.
+  //
+  // The thinking is that validation as a result of onChange and onBlur
+  // should never block user input. Note: This method should never be called
+  // during the submission phase because validation prior to submission
+  // is actaully high-priority since we absolutely need to guarantee the
+  // form is valid before executing props.onSubmit.
+  const validateFormWithLowPriority = React.useCallback(
+    (values: Values = state.values) => {
+      if (!hasValidationFns) {
+        // No validation functions, so we bail out immediately
+        return Promise.resolve(emptyErrors);
+      }
 
-  React.useEffect(() => {
-    if (
-      prevState.touched !== state.touched &&
-      !!validateOnBlur &&
-      !state.isSubmitting &&
-      isMounted.current != null
-    ) {
-      const [validate, cancel] = makeCancelable(validateForm());
-      validate.then(x => x).catch(x => x); // catch the rejection silently
-      return cancel;
-    }
-    return;
-  }, [
-    prevState.touched,
-    state.isSubmitting,
-    state.touched,
-    validateForm,
-    validateOnBlur,
-    isMounted,
-  ]);
+      return unstable_runWithPriority(LowPriority, () => {
```

Comment:
@aweary is this correctly implemented? My thinking is that form validation in response to user input is less important than the actual input, and thus should be marked as low-pri (it is a decent candidate for debounce). Do I need to also mark the update to value in response to `onChange` as Immediate or User Blocking?

Reply:
I confirmed this use case with Brandon in person.